conda_forge_tick/auto_tick.py

```diff
@@ -477,6 +482,64 @@ def migration_factory(
         )
 
 
+def _outside_pin_range(pin_spec, current_pin, new_version):
+    pin_level = len(pin_spec.split('.'))
+    current_split = current_pin.split('.')
+    new_split = new_version.split('.')
+    # if our pin spec is more exact than our current pin then rebuild more precisely
+    if pin_level > len(current_split):
+        return True
+    for i in range(pin_level):
+        if current_split[i] != new_split[i]:
+            return True
+    return False
+
+
+def create_migration_yaml_creator(migrators: MutableSequence[Migrator], gx: nx.DiGraph):
+    with indir("../conda-forge-pinning-feedstock/recipe"):
+        pinnings = parse_config_file('conda_build_config.yaml',
+                                     config=Config(**CB_CONFIG))
+    for k, package_pin_list in pinnings.items():
+        # exclude non-package keys
+        if k not in gx.nodes:
+            # conda_build_config.yaml can't have `-` unlike our package names
+            k = k.replace('_', '-')
+
+        if (k in gx.nodes) and not gx.nodes[k]['payload'].get('archived', False) and gx.nodes[k]['payload'].get('version'):
+
+            current_pins = list(map(str, package_pin_list))
+            current_version = str(gx.nodes[k]['payload']['version'])
+
+            # XXX: uncomment following line and remove the one after once graph
+            # has been re-loaded with the pins properly
+
+            # meta_yaml = gx.nodes[k]['payload']['meta_yaml']
+            meta_yaml = parse_meta_yaml(gx.nodes[k]['payload']['raw_meta_yaml'])
+
+            # find the most stringent max pin for this feedstock if any
+            pin_spec = ''
+            for block in [meta_yaml] + meta_yaml.get("outputs", []) or []:
+                build = block.get('build', {}) or {}
+                max_pin = next(iter(build.get('run_exports', [''])))
```

Comment:
What is this line supposed to do? I am messing around locally and not understanding what is does in various cases.